lib/states/wi/madison.R

```diff
@@ -8,86 +8,136 @@ source("common.R")
-    subject_sex = tr_sex[Sex]
-  ) %>%
-  helpers$add_lat_lng(
-  ) %>%
-  helpers$add_shapefiles_data(
-  ) %>%
-  # NOTE: shapefiles don't appear to include district 2 and accompanying
-  # sectors
-  rename(
-    sector = Sector,
-    district = District,
-    raw_race = Race
-  ) %>%
-  standardize(d$metadata)
+    mutate(
+      vehicle_registration_state = coalesce(`Reg State`, State),
+      vehicle_year = coalesce(`Model Year`, Year),
+    ) %>%
+    merge_rows(
+      Date,
+      Time,
+      onStreet,
+      onStreetName,
+      OfficerName,
+      Race,
+      Sex,
+      Make,
+      Model,
+      vehicle_year,
+      vehicle_registration_state,
+      Limit,
+      OverLimit,
+      Type
+    ) %>%
+    rename(
+      ticketed = Type, 
+      violation = `Statute Description`,
+      vehicle_make = Make,
+      vehicle_model = Model,
+      vehicle_color = Color,
+      posted_speed = Limit,
+      subject_age = Age, 
+      time = Time
+    ) %>%
+    separate_cols(
+      OfficerName = c("officer_last_name", "officer_first_name")
+    ) %>%
+    mutate(
+      # OLD NOTE: Statute Descriptions are almost all vehicular, there are a few
+      # pedestrian related Statute Descriptions, but it's unclear whether
+      # the pedestrian or vehicle is failing to yield, but this represents a
+      # quarter of a percent maximum
+      # UPDATED NOTE: Similarly, there are a few pedestrian related Statute
+      # Descriptions, but they are extremely infrequent (~ < a quarter of a percent)
+      type = "vehicular",
+      speed = as.integer(posted_speed) + as.integer(OverLimit),
+      date = parse_date(Date, "%Y/%m/%d"),
+      location = coalesce(onStreet, onStreetName),
+      # logic: for old data, if ticket # is NA, it's a warning
+      # for new data, if ticketed == "Warning", it's a warning
+      # new data doesn't have ticket # var; old data doesn't have ticketed var
+      # so if ticketed is NA and Ticket # is NA (i.e., from the old data), it's a warning
+      # if ticketed == "Warning", it's a warning
+      # if ticketed == "ELCI", its a ticket
+      # if Ticket # is not NA, (i.e., from the new data and not a warning) it's a ticket
+      # warning formula: ticketed == "Warning" OR (ticketed is NA & Ticket # is NA)
+      warning_issued = case_when(
+        ticketed == "Warning" ~ TRUE,
+        is.na(ticketed) & is.na(`Ticket #`) ~ TRUE,
+        # if anything else is the case, then its a ticket
+        TRUE ~ FALSE
+      ),
+      citation_issued = !warning_issued,
+      # TODO(phoebe): can we get arrests?
+      # https://app.asana.com/0/456927885748233/595493946182543
+      outcome = first_of(
+        citation = citation_issued,
+        warning = warning_issued
+      ),
+      # there are several thousand rows in the 
+      # new data where race and sex are switched
+      fixed_race = ifelse(Race %in% c("M","F"), Sex, Race), 
+      fixed_sex = ifelse(Race %in% c("M","F"), Race, Sex), 
```

Comment:
In most cases, it's helpful to stick to `if_else` to make sure it's typed.

Reply:
good catch, thanks!